File: src/main/java/com/kishlaly/ta/analyze/tasks/blocks/groups/ThreeDisplays_Buy_EFI_1.java
package com.kishlaly.ta.analyze.tasks.blocks.groups;

import com.kishlaly.ta.analyze.tasks.ThreeDisplays;
import com.kishlaly.ta.analyze.tasks.blocks.TaskBlockJava;
import com.kishlaly.ta.analyze.tasks.blocks.commons.ScreenBasicValidation;

import java.util.ArrayList;
import java.util.List;

/**
 * The first screen is not used
 * The second screen checks the three EFI values:
 * + third from the end is negative
 * + penultimate and last are positive
 * <p>
 * SL sliding on the average Bollinger band or TP at the top of the channel, if the last quote is not very high
 * <p>
 * !!! Can't flip for shorts !!!
 */
//TODO finish
public class ThreeDisplays_Buy_EFI_1 implements BlocksGroupJava {
    @Override
    public List<TaskBlockJava> blocks() {
        ThreeDisplays.Config.STOCH_CUSTOM = 30;

        return new ArrayList<TaskBlockJava>() {{
            add(new ScreenBasicValidation());

        }};

    }

    @Override
    public String comments() {
        return "EFI rose above zero and two values are positive";
    }
}

File: src/main/kotlin/com/kishlaly/ta/analyze/tasks/groups/threedisplays/ThreeDisplays_Buy_EFI_1.kt
package com.kishlaly.ta.analyze.tasks.groups.threedisplays

import com.kishlaly.ta.analyze.tasks.blocks.TaskBlock
import com.kishlaly.ta.analyze.tasks.blocks.commons.ScreenBasicValidation
import com.kishlaly.ta.analyze.tasks.blocks.two.Long_ScreenTwo_EFI_ThirdNegative
import com.kishlaly.ta.analyze.tasks.blocks.two.Long_ScreenTwo_EFI_TwoPositive
import com.kishlaly.ta.analyze.tasks.groups.AbstractBlocksGroup

/**
 * The first screen is not used
 * The second screen checks the three EFI values:
 * + third from the end is negative
 * + penultimate and last are positive
 * <p>
 * SL sliding on the average Bollinger band or TP at the top of the channel, if the last quote is not very high
 * <p>
 * !!! Can't flip for shorts !!!
 */
class ThreeDisplays_Buy_EFI_1 : AbstractBlocksGroup() {

    override fun blocks(): List<TaskBlock> {
        return object : ArrayList<TaskBlock>() {
            init {
                add(ScreenBasicValidation())

                add(Long_ScreenTwo_EFI_ThirdNegative())
                add(Long_ScreenTwo_EFI_TwoPositive())
            }
        }
    }

    override fun comments() = super.comments() + "EFI rose above zero and two values are positive"

}